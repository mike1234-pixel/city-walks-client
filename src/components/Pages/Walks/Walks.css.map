{"version":3,"sourceRoot":"","sources":["Walks.scss"],"names":[],"mappings":"AAAA;AAEA;EACE;;;AAGF;EACE;;;AAGF;EACE;;;AAGF;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;;AAGF;EACE;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;;AAGF;EACE;EACA;EACA;EACA;;;AAGF;EACE;EACA;EACA;EACA;;;AAIA;AAAA;EAEE;EACA;EACA","file":"Walks.css"}