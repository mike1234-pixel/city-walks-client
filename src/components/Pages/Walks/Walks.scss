/** @format */

.walk-card {
  transition: transform 0.2s; /* Animation */
}

.walk-card:hover {
  transform: scale(1.05);
}

.walks-search-icon {
  margin: 25px 0;
}

.pagination-buttons {
  margin-top: 60px;
  height: 40px;
  list-style: none;
  display: flex;
  justify-content: center;
  margin-top: 60px;
  display: flex;
  padding-left: 0;
  list-style: none;
  border-radius: 0.25rem;
}

.pagination-buttons a {
  padding: 10px;
  margin: 8px;
  border-radius: 2px;
  cursor: pointer;
  font-size: 0.9rem;
  color: #212529;
  background-color: transparent;
  border: 0;
  outline: 0;
  transition: all 0.3s linear;
}

.pagination-buttons a:hover {
  color: white;
  background: var(--blue);
  box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.16), 0 2px 10px 0 rgba(0, 0, 0, 0.12);
  transition: all 0.2s linear;
}

.pagination-buttons a:active {
  color: #fff;
  background-color: var(--blue);
  box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.16), 0 2px 10px 0 rgba(0, 0, 0, 0.12);
  transition: all 0.2s linear;
}

.pagination-active {
  a,
  a:hover {
    background: var(--blue);
    color: white;
    border-radius: 2px;
  }
}
