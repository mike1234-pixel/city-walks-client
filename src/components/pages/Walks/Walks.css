.walk-card {
  transition: transform 0.2s; /* Animation */
}

.walk-card:hover {
  transform: scale(1.05);
}

.walks-search-icon {
  margin: 25px 0;
}
