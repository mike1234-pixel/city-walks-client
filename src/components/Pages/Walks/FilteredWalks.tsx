import React, { ReactNode } from "react";
import { MDBContainer, MDBBtn } from "mdbreact";
import ReactPaginate from "react-paginate";
import { FaSearchLocation } from "react-icons/fa";
import { setSearchValue } from "../../../actions/actions";
import Walk from "../../../types/Walks/Walk";
import WalkCard from "./WalkCard";

interface FilteredResultsProps {
  searchValue: string;
  walks: Array<Walk>;
  pageCount: number;
  pagesVisited: number;
  walksPerPage: number;
  changePage: (selected: number) => void;
}

const FilteredResults = (props: FilteredResultsProps) => {
  const {
    searchValue,
    walks,
    pageCount,
    pagesVisited,
    walksPerPage,
    changePage,
  } = props;

  if (searchValue === "") {
    return (
      <MDBContainer>
        <div className="page">
          <div className="page-heading-container">
            <h1 className="page-heading">Walks</h1>
            <MDBBtn
              outline
              color="elegant"
              className="city-card-btn"
              onClick={() => setSearchValue("")}
            >
              Show all walks
            </MDBBtn>
            <p data-testid="walks-search-icon" className="walks-search-icon">
              <FaSearchLocation className="search-location-icon" />
              {`  ${searchValue}`}
            </p>
          </div>
          <div className="card-container">
            {walks
              .slice(pagesVisited, pagesVisited + walksPerPage)
              .map((walk) => {
                const {
                  _id,
                  walk: walkName,
                  city,
                  description,
                  coverImg,
                } = walk;

                return (
                  <div key={_id}>
                    <WalkCard
                      id={_id}
                      name={walkName}
                      city={city}
                      description={description}
                      imgSrc={coverImg}
                    />
                  </div>
                );
              })}
          </div>
          <ReactPaginate
            previousLabel={"Previous"}
            nextLabel={"Next"}
            pageCount={pageCount}
            onPageChange={changePage}
            containerClassName={"pagination-buttons"}
            previousLinkClassName={"previous-button"}
            nextLinkClassName={"next-button"}
            disabledClassName={"pagination-disabled"}
            activeClassName={"pagination-active"}
          />
        </div>
      </MDBContainer>
    );
  }

  return (
    <MDBContainer>
      <div className="page">
        <div className="page-heading-container">
          <h1 className="page-heading" data-testid="walks-page-heading">
            Walks
          </h1>
          <MDBBtn
            outline
            color="elegant"
            className="city-card-btn"
            onClick={() => setSearchValue("")}
          >
            Show all walks
          </MDBBtn>
          <p data-testid="walks-search-icon" className="walks-search-icon">
            <FaSearchLocation className="search-location-icon" />
            {`  ${searchValue}`}
          </p>
        </div>
        <div className="card-container">
          {walks.map((walk: Walk) => {
            const { _id, walk: walkName, city, coverImg, description } = walk;

            if (
              walkName.toLowerCase().includes(searchValue.toLowerCase()) ||
              city.toLowerCase().includes(searchValue.toLowerCase())
            ) {
              return (
                <div key={_id}>
                  <WalkCard
                    id={_id}
                    city={city}
                    name={walkName}
                    description={description}
                    imgSrc={coverImg}
                  />
                </div>
              );
            }
          })}
        </div>
      </div>
    </MDBContainer>
  );
};

export default FilteredResults;
