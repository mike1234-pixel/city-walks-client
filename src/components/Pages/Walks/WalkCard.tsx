import {
  MDBCard,
  MDBCardImage,
  MDBCardBody,
  MDBCardTitle,
  MDBCardText,
  MDBBtn,
} from "mdbreact";
import { Link } from "react-router-dom";
import { GiWalkingBoot } from "react-icons/gi";
import urlify from "../../../utils/urlify";
import "./Walks.css";

interface WalkCardProps {
  id: string;
  imgSrc: string;
  name: string;
  city: string;
  description: string;
}

const WalkCard = (props: WalkCardProps) => {
  const { id, imgSrc, name, city, description } = props;

  return (
    <div key={id}>
      <Link to={"walks/" + urlify(name)}>
        <MDBCard className='walk-card'>
          <MDBCardImage
            className='cutter img-fluid'
            src={imgSrc}
            alt={name}
            waves
          />
          <MDBCardBody>
            <MDBCardTitle>{city}</MDBCardTitle>
            <MDBCardTitle className='display-font'>{name}</MDBCardTitle>
            <MDBCardText>{description}</MDBCardText>
            <MDBBtn outline color='elegant' className='city-card-btn'>
              Explore <GiWalkingBoot />
            </MDBBtn>
          </MDBCardBody>
        </MDBCard>
      </Link>
    </div>
  );
};

export default WalkCard;
