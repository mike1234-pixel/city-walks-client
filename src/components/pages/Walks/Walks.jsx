import { useEffect, useContext } from "react"
import { MDBIcon, MDBBtn, MDBCard, MDBCardBody, MDBCardImage, MDBCardTitle, MDBCardText, MDBCol } from 'mdbreact'
import { FaSearchLocation } from "react-icons/fa"
import { SearchContext } from '../../../context/SearchContext'
import urlify from '../../../functions/urlify'
import { Link } from "react-router-dom"
import PopUp from "../../PopUp/PopUp"
import { LoginContext } from "../../../context/LoginContext"
import './Walks.css'

const Walks = (props) => {

    const walksArr = props.walks

    let { searchValue, setSearchValue } = useContext(SearchContext)
    const { popupVisible } = useContext(LoginContext)

    useEffect(() => {
        window.scrollTo(0, 0);
      });
    
    if (searchValue === "") {
    return (
        <div className="min-page-height">
            {popupVisible && <PopUp/>}
            <div className="page-heading-container">
                <h1 className="page-heading">Walks</h1>
                <MDBBtn outline color="elegant" className="city-card-btn" onClick={() => setSearchValue("")}>Show all walks</MDBBtn>
                <p data-testid="walks-search-icon" className="walks-search-icon"><FaSearchLocation className="search-location-icon"/>{`  ${searchValue}`}</p>
            </div>
            <div  className="card-container">
            {walksArr.map(v => {
                return (
                    <MDBCol key={v._id}>
                    <MDBCard className="walk-card">
                        <MDBCardImage className="cutter img-fluid" src={v.coverImg} alt={v.walk} waves/>
                        <MDBCardBody>
                        <MDBCardTitle>{v.city}</MDBCardTitle>
                        <MDBCardTitle className="display-font">{v.walk}</MDBCardTitle>
                        <MDBCardText>{v.description}</MDBCardText>
                        <Link to={'walks/'+urlify(v.walk)}><MDBBtn outline color="elegant" className="city-card-btn">Explore <MDBIcon icon="walking" /></MDBBtn></Link>
                        </MDBCardBody>
                    </MDBCard>
                    </MDBCol>
                    )
                })}
            </div>
        </div>
    )
    } else if (walksArr.map(v => { v.walk.toLowerCase().includes(searchValue.toLowerCase()) || v.city.toLowerCase().includes(searchValue.toLowerCase()) } )) {
        return (
        <div className="min-page-height">
            <div className="page-heading-container">
                <h1 className="page-heading" data-testid="walks-page-heading">Walks</h1>
                <MDBBtn outline color="elegant" className="city-card-btn" onClick={() => setSearchValue("")}>Show all walks</MDBBtn>
                <p data-testid="walks-search-icon" className="walks-search-icon"><FaSearchLocation className="search-location-icon"/>{`  ${searchValue}`}</p>
            </div>
            <div className="card-container">
                {walksArr.map(v => {
                if (v.walk.toLowerCase().includes(searchValue.toLowerCase()) || v.city.toLowerCase().includes(searchValue.toLowerCase())) {
                return (
                    <MDBCol key={v._id}>
                    <MDBCard className="walk-card">
                        <MDBCardImage className="cutter img-fluid" src={v.coverImg} alt={v.walk} waves/>
                        <MDBCardBody>
                        <MDBCardTitle>{v.city}</MDBCardTitle>
                        <MDBCardTitle className="display-font">{v.walk}</MDBCardTitle>
                        <MDBCardText>{v.description}</MDBCardText>
                        <Link to={'walks/'+urlify(v.walk)}><MDBBtn outline color="elegant" className="city-card-btn">Explore <MDBIcon icon="walking" /></MDBBtn></Link>
                        </MDBCardBody>
                    </MDBCard>
                    </MDBCol>
                    )
                }} 
                )}
            </div>
        </div>
        )
    }
}

export default Walks