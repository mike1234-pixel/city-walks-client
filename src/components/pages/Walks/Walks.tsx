import React, { ReactNode, useEffect, useState } from "react"
import { MDBBtn, MDBCard, MDBCardBody, MDBCardImage, MDBCardTitle, MDBCardText, MDBContainer } from 'mdbreact'
import { FaSearchLocation } from "react-icons/fa"
import urlify from '../../../functions/urlify'
import { Link } from "react-router-dom"
import ReactPaginate from "react-paginate"
import { GiWalkingBoot } from 'react-icons/gi'
import { connect } from 'react-redux';
import Walk from "../../../types/Walks/Walk"
import GlobalState from "../../../types/State/Global/State"
import './Walks.scss'

interface Props {
    searchValue: string;
    setSearchValue: Function;
    walks: Array<Walk>;
}

const Walks: React.FC<any> = (props: Props) => {

    const { searchValue, setSearchValue, walks } = props;

    // display all walks pagination

    const [pageNumber, setPageNumber] = useState<number>(0)
    let filteredResults: ReactNode;

    const walksPerPage: number = 3;
    const pagesVisited: number = pageNumber * walksPerPage;

    const pageCount: number = Math.ceil(walks.length / walksPerPage);

    const displayAllWalks: () => Array<ReactNode> = () => {
        return (
            walks.slice(pagesVisited, pagesVisited + walksPerPage).map(v => {
                return (
                    <div key={v._id}>
                        <Link to={'walks/' + urlify(v.walk)}>
                            <MDBCard className="walk-card">
                                <MDBCardImage className="cutter img-fluid" src={v.coverImg} alt={v.walk} waves />
                                <MDBCardBody>
                                    <MDBCardTitle>{v.city}</MDBCardTitle>
                                    <MDBCardTitle className="display-font">{v.walk}</MDBCardTitle>
                                    <MDBCardText>{v.description}</MDBCardText>
                                    <MDBBtn outline color="elegant" className="city-card-btn">Explore <GiWalkingBoot /></MDBBtn>
                                </MDBCardBody>
                            </MDBCard>
                        </Link>
                    </div>
                )
            })
        )
    }

    const changePage = ({ selected }: { selected: number }) => {
        setPageNumber(selected);
    }

    useEffect(() => {
        window.scrollTo(0, 0);
    });

    if (searchValue === "") {
        filteredResults =
            <MDBContainer>
                <div className="min-page-height">
                    <div className="page-heading-container">
                        <h1 className="page-heading">Walks</h1>
                        <MDBBtn outline color="elegant" className="city-card-btn" onClick={() => setSearchValue("")}>Show all walks</MDBBtn>
                        <p data-testid="walks-search-icon" className="walks-search-icon"><FaSearchLocation className="search-location-icon" />{`  ${searchValue}`}</p>
                    </div>
                    <div className="card-container">
                        {displayAllWalks()}
                    </div>
                    <ReactPaginate
                        previousLabel={"Previous"}
                        nextLabel={"Next"}
                        pageCount={pageCount}
                        onPageChange={changePage}
                        containerClassName={"pagination-buttons"}
                        previousLinkClassName={"previous-button"}
                        nextLinkClassName={"next-button"}
                        disabledClassName={"pagination-disabled"}
                        activeClassName={"pagination-active"}
                    />
                </div>
            </MDBContainer>

    } else if (walks.map((v: Walk) => { v.walk.toLowerCase().includes(searchValue.toLowerCase()) || v.city.toLowerCase().includes(searchValue.toLowerCase()) })) {
        filteredResults =
            <MDBContainer>
                <div className="min-page-height">
                    <div className="page-heading-container">
                        <h1 className="page-heading" data-testid="walks-page-heading">Walks</h1>
                        <MDBBtn outline color="elegant" className="city-card-btn" onClick={() => setSearchValue("")}>Show all walks</MDBBtn>
                        <p data-testid="walks-search-icon" className="walks-search-icon"><FaSearchLocation className="search-location-icon" />{`  ${searchValue}`}</p>
                    </div>
                    <div className="card-container">
                        {walks.map((v: Walk) => {
                            if (v.walk.toLowerCase().includes(searchValue.toLowerCase()) || v.city.toLowerCase().includes(searchValue.toLowerCase())) {
                                return (
                                    <div key={v._id}>
                                        <Link to={'walks/' + urlify(v.walk)}>
                                            <MDBCard className="walk-card">
                                                <MDBCardImage className="cutter img-fluid" src={v.coverImg} alt={v.walk} waves />
                                                <MDBCardBody>
                                                    <MDBCardTitle>{v.city}</MDBCardTitle>
                                                    <MDBCardTitle className="display-font">{v.walk}</MDBCardTitle>
                                                    <MDBCardText>{v.description}</MDBCardText>
                                                    <MDBBtn outline color="elegant" className="city-card-btn">Explore <GiWalkingBoot /></MDBBtn>
                                                </MDBCardBody>
                                            </MDBCard>
                                        </Link>
                                    </div>
                                )
                            }
                        }
                        )}
                    </div>
                </div>
            </MDBContainer>

    }

    return (
        <div>
            {filteredResults}
        </div>
    )
}

const mapStateToProps = (state: GlobalState) => ({
    walks: state.walksState.walks,
    searchValue: state.searchState.searchValue,
});

const mapDispatchToProps = (dispatch: Function) => {
    return {
        setSearchValue: (inputValue: string) => dispatch({ type: 'HANDLE_CHANGE_SEARCH', inputValue })
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Walks);