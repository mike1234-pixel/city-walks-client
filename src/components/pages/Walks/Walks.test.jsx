import renderer from 'react-test-renderer'
import { SearchContextProvider } from "../../../context/SearchContext"
import { LoginContextProvider } from "../../../context/LoginContext"
import WalksTestData from "../../../container/WalksTestData.json"
import { BrowserRouter } from 'react-router-dom';
import Walks from './Walks'

// snapshot test
it('Walks Page renders correctly', () => {  

  const walks = WalksTestData

  const tree = renderer
    .create(
    <SearchContextProvider>
      <LoginContextProvider>
        <BrowserRouter>
          <Walks walks={walks}/>
        </BrowserRouter>
      </LoginContextProvider>
    </SearchContextProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});