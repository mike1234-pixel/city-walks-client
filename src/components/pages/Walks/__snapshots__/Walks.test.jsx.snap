// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Walks Page renders correctly 1`] = `
<div
  className="min-page-height"
>
  <div
    className="page-heading-container"
  >
    <h1
      className="page-heading"
    >
      Walks
    </h1>
    <button
      className="btn-elegant btn-outline-elegant btn Ripple-parent city-card-btn"
      data-test="button"
      onClick={[Function]}
      onMouseUp={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      Show all walks
      <div
        className="Ripple Ripple-outline "
        data-test="waves"
        style={
          Object {
            "height": "0px",
            "left": "0px",
            "top": "0px",
            "width": "0px",
          }
        }
      />
    </button>
    <p
      className="walks-search-icon"
      data-testid="walks-search-icon"
    >
      <svg
        className="search-location-icon"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 512 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M505.04 442.66l-99.71-99.69c-4.5-4.5-10.6-7-17-7h-16.3c27.6-35.3 44-79.69 44-127.99C416.03 93.09 322.92 0 208.02 0S0 93.09 0 207.98s93.11 207.98 208.02 207.98c48.3 0 92.71-16.4 128.01-44v16.3c0 6.4 2.5 12.5 7 17l99.71 99.69c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.59.1-33.99zm-297.02-90.7c-79.54 0-144-64.34-144-143.98 0-79.53 64.35-143.98 144-143.98 79.54 0 144 64.34 144 143.98 0 79.53-64.35 143.98-144 143.98zm.02-239.96c-40.78 0-73.84 33.05-73.84 73.83 0 32.96 48.26 93.05 66.75 114.86a9.24 9.24 0 0 0 14.18 0c18.49-21.81 66.75-81.89 66.75-114.86 0-40.78-33.06-73.83-73.84-73.83zm0 96c-13.26 0-24-10.75-24-24 0-13.26 10.75-24 24-24s24 10.74 24 24c0 13.25-10.75 24-24 24z"
        />
      </svg>
        
    </p>
  </div>
  <div
    className="card-container"
  >
    <div
      className="col"
      data-test="col"
    >
      <div
        className="card walk-card"
        data-test="card"
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="Ripple-parent"
            onMouseDown={[Function]}
            style={
              Object {
                "touchAction": "unset",
              }
            }
          >
            <img
              alt="Fulham"
              className="cutter img-fluid"
              data-test="card-image"
              src="https://upload.wikimedia.org/wikipedia/commons/thumb/5/5b/Fulham_Palace_courtyard_-_geograph.org.uk_-_835758.jpg/360px-Fulham_Palace_courtyard_-_geograph.org.uk_-_835758.jpg"
            />
            <div
              className="mask rgba-white-slight"
              data-test="mask"
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </div>
        </div>
        <div
          className="card-body"
          data-test="card-body"
        >
          <h4
            className="card-title"
            data-test="card-title"
          >
            London
          </h4>
          <h4
            className="card-title display-font"
            data-test="card-title"
          >
            Fulham
          </h4>
          <p
            className="card-text"
            data-test="card-text"
          >
            Fulham is an upmarket residential area known for trendy shopping streets like Kingâ€™s Road and Fulham Road.
          </p>
          <a
            href="/walks/fulham"
            onClick={[Function]}
          >
            <button
              className="btn-elegant btn-outline-elegant btn Ripple-parent city-card-btn"
              data-test="button"
              onMouseUp={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              Explore 
              <i
                className="fa fa-walking"
                data-test="fa"
              />
              <div
                className="Ripple Ripple-outline "
                data-test="waves"
                style={
                  Object {
                    "height": "0px",
                    "left": "0px",
                    "top": "0px",
                    "width": "0px",
                  }
                }
              />
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      className="col"
      data-test="col"
    >
      <div
        className="card walk-card"
        data-test="card"
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="Ripple-parent"
            onMouseDown={[Function]}
            style={
              Object {
                "touchAction": "unset",
              }
            }
          >
            <img
              alt="Richmond"
              className="cutter img-fluid"
              data-test="card-image"
              src="https://upload.wikimedia.org/wikipedia/commons/9/98/View_from_Richmond_Hill_Surrey_01.jpg"
            />
            <div
              className="mask rgba-white-slight"
              data-test="mask"
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </div>
        </div>
        <div
          className="card-body"
          data-test="card-body"
        >
          <h4
            className="card-title"
            data-test="card-title"
          >
            London
          </h4>
          <h4
            className="card-title display-font"
            data-test="card-title"
          >
            Richmond
          </h4>
          <p
            className="card-text"
            data-test="card-text"
          >
            The latter half of this long but stunning river walk is not accessible at high tide, so check the tide tables at https://thames-path.com
          </p>
          <a
            href="/walks/richmond"
            onClick={[Function]}
          >
            <button
              className="btn-elegant btn-outline-elegant btn Ripple-parent city-card-btn"
              data-test="button"
              onMouseUp={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              Explore 
              <i
                className="fa fa-walking"
                data-test="fa"
              />
              <div
                className="Ripple Ripple-outline "
                data-test="waves"
                style={
                  Object {
                    "height": "0px",
                    "left": "0px",
                    "top": "0px",
                    "width": "0px",
                  }
                }
              />
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      className="col"
      data-test="col"
    >
      <div
        className="card walk-card"
        data-test="card"
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="Ripple-parent"
            onMouseDown={[Function]}
            style={
              Object {
                "touchAction": "unset",
              }
            }
          >
            <img
              alt="Camden"
              className="cutter img-fluid"
              data-test="card-image"
              src="https://upload.wikimedia.org/wikipedia/commons/6/6c/Camden_Town_9.jpg"
            />
            <div
              className="mask rgba-white-slight"
              data-test="mask"
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </div>
        </div>
        <div
          className="card-body"
          data-test="card-body"
        >
          <h4
            className="card-title"
            data-test="card-title"
          >
            London
          </h4>
          <h4
            className="card-title display-font"
            data-test="card-title"
          >
            Camden
          </h4>
          <p
            className="card-text"
            data-test="card-text"
          >
            Camden is notorious for its lively markets, which surge with thousands of shoppers - most of them pierced - each weekend.
          </p>
          <a
            href="/walks/camden"
            onClick={[Function]}
          >
            <button
              className="btn-elegant btn-outline-elegant btn Ripple-parent city-card-btn"
              data-test="button"
              onMouseUp={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              Explore 
              <i
                className="fa fa-walking"
                data-test="fa"
              />
              <div
                className="Ripple Ripple-outline "
                data-test="waves"
                style={
                  Object {
                    "height": "0px",
                    "left": "0px",
                    "top": "0px",
                    "width": "0px",
                  }
                }
              />
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      className="col"
      data-test="col"
    >
      <div
        className="card walk-card"
        data-test="card"
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="Ripple-parent"
            onMouseDown={[Function]}
            style={
              Object {
                "touchAction": "unset",
              }
            }
          >
            <img
              alt="Wimbledon Common"
              className="cutter img-fluid"
              data-test="card-image"
              src="https://images.unsplash.com/photo-1595514824941-32dddb85a383?ixid=MXwxMjA3fDB8MHxzZWFyY2h8N3x8d2ltYmxlZG9ufGVufDB8fDB8&ixlib=rb-1.2.1&auto=format&fit=crop&w=600&q=60"
            />
            <div
              className="mask rgba-white-slight"
              data-test="mask"
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </div>
        </div>
        <div
          className="card-body"
          data-test="card-body"
        >
          <h4
            className="card-title"
            data-test="card-title"
          >
            London
          </h4>
          <h4
            className="card-title display-font"
            data-test="card-title"
          >
            Wimbledon Common
          </h4>
          <p
            className="card-text"
            data-test="card-text"
          >
            Home to a windmill and the famous tennis complex, Wimbledon is worth a visit even when there's no one on Centre Court.
          </p>
          <a
            href="/walks/wimbledon-common"
            onClick={[Function]}
          >
            <button
              className="btn-elegant btn-outline-elegant btn Ripple-parent city-card-btn"
              data-test="button"
              onMouseUp={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              Explore 
              <i
                className="fa fa-walking"
                data-test="fa"
              />
              <div
                className="Ripple Ripple-outline "
                data-test="waves"
                style={
                  Object {
                    "height": "0px",
                    "left": "0px",
                    "top": "0px",
                    "width": "0px",
                  }
                }
              />
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      className="col"
      data-test="col"
    >
      <div
        className="card walk-card"
        data-test="card"
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="Ripple-parent"
            onMouseDown={[Function]}
            style={
              Object {
                "touchAction": "unset",
              }
            }
          >
            <img
              alt="Embankment"
              className="cutter img-fluid"
              data-test="card-image"
              src="https://images.pexels.com/photos/415992/pexels-photo-415992.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260"
            />
            <div
              className="mask rgba-white-slight"
              data-test="mask"
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </div>
        </div>
        <div
          className="card-body"
          data-test="card-body"
        >
          <h4
            className="card-title"
            data-test="card-title"
          >
            London
          </h4>
          <h4
            className="card-title display-font"
            data-test="card-title"
          >
            Embankment
          </h4>
          <p
            className="card-text"
            data-test="card-text"
          >
            Stroll from the Thames past the finest secondhand bookshops and into the nation's Portrait Gallery.
          </p>
          <a
            href="/walks/embankment"
            onClick={[Function]}
          >
            <button
              className="btn-elegant btn-outline-elegant btn Ripple-parent city-card-btn"
              data-test="button"
              onMouseUp={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              Explore 
              <i
                className="fa fa-walking"
                data-test="fa"
              />
              <div
                className="Ripple Ripple-outline "
                data-test="waves"
                style={
                  Object {
                    "height": "0px",
                    "left": "0px",
                    "top": "0px",
                    "width": "0px",
                  }
                }
              />
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      className="col"
      data-test="col"
    >
      <div
        className="card walk-card"
        data-test="card"
      >
        <div
          className="view"
          data-test="view"
          onMouseDown={[Function]}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="Ripple-parent"
            onMouseDown={[Function]}
            style={
              Object {
                "touchAction": "unset",
              }
            }
          >
            <img
              alt="Crouch End"
              className="cutter img-fluid"
              data-test="card-image"
              src="https://upload.wikimedia.org/wikipedia/commons/e/e5/Crouch_End_20_Sept_2011.jpg"
            />
            <div
              className="mask rgba-white-slight"
              data-test="mask"
            />
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </div>
        </div>
        <div
          className="card-body"
          data-test="card-body"
        >
          <h4
            className="card-title"
            data-test="card-title"
          >
            London
          </h4>
          <h4
            className="card-title display-font"
            data-test="card-title"
          >
            Crouch End
          </h4>
          <p
            className="card-text"
            data-test="card-text"
          >
            A quiet nature trail and a funky neighbourhood full of boutique shops await those who make this journey to North London.
          </p>
          <a
            href="/walks/crouch-end"
            onClick={[Function]}
          >
            <button
              className="btn-elegant btn-outline-elegant btn Ripple-parent city-card-btn"
              data-test="button"
              onMouseUp={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              Explore 
              <i
                className="fa fa-walking"
                data-test="fa"
              />
              <div
                className="Ripple Ripple-outline "
                data-test="waves"
                style={
                  Object {
                    "height": "0px",
                    "left": "0px",
                    "top": "0px",
                    "width": "0px",
                  }
                }
              />
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
